File: data/gcc_versions_comparison/src/program/src/func41.c
#include "program.h" 
array_t* func41(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array13;
   if (pCounter > 0) {
      array13 = vars->data[--pCounter];
      array13->refC++;
      DEBUG_COPY(array13->id);
   } else {
      array13 = (array_t*)malloc(sizeof(array_t));
      array13->size = 370;
      array13->refC = 1;
      array13->id = 13;
      array13->data = (unsigned int*)malloc(array13->size*sizeof(unsigned int));
      memset(array13->data, 0, array13->size*sizeof(unsigned int));
      DEBUG_NEW(array13->id);
   }
   unsigned int loop7 = 0;
   unsigned int loopLimit7 = (rand()%loopsFactor)/5 + 1;
   for(; loop7 < loopLimit7; loop7++) {
      for (int i = 0; i < array13->size; i++) {
         array13->data[i]++;
      }
   }
   for (int i = 0; i < array13->size; i++) {
      if (array13->data[i] == 91) { 
         return array13;
      }
   }
   return array13;
}

File: data/multilang_comparison/program_go/src/func41.go
package main
import "math/rand"
var _ = rand.Intn
func Func41(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array13 *Array
   if pCounter > 0 {
       array13 = vars.Data[pCounter-1]
       pCounter--
       array13.RefC++
   } else {
       array13 = &Array{
           Size: 370,
           RefC: 1,
           Id: 13,
           Data: make([]uint32, 370),
       }
   }
   var loop7 int = 0;
   var loopLimit7 int = (50)/5 + 1;
   for ; loop7 < loopLimit7; loop7++ {
      for i := 0; i < array13.Size; i++ {
          array13.Data[i]++
      }
   }
   for i := 0; i < array13.Size; i++ {
       if array13.Data[i] == 91 {
           return array13
       }
   }
   return array13;
}

